### PR TITLE
[SPARKR-153] phase 1: implement fold() and aggregate().

### PR DESCRIPTION
**Pull request by sun-rui**
aggregate() is renamed to aggregateRDD() due to name conflict with the built-in function aggregate() in R.

### CODE REVIEW

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -1374,6 +1374,71 @@ setMethod("top",
...
+#' 
+#' @param rdd An RDD.
+#' @param zeroValue A neutral "zero value".
+#' @param seqOp A function to aggregate the RDD elements. It may return a different
+#'              result type from the type of the RDD elements.
+#' @param combOp A function to aggregate results of seqOp.
+#' @return The aggregation result.
+#' @rdname aggregateRDD
+#' @export
+#' @examples
```
Comment (shivaram) [inline pkg/R/RDD.R]: It'll be good to add a `@seealso reduce` or to compare fold, reduce, aggregate.

**Comment by shivaram** [comment]
Thanks @sun-rui - This looks pretty good. 

@concretevitamin -- Any other comments ?

File: pkg/inst/tests/test_rdd.R [hunk 2]
```diff
@@ -302,6 +302,28 @@ test_that("top() on RDDs", {
...
+test_that("fold() on RDDs", {
+  actual <- fold(rdd, 0, "+")
+  expect_equal(actual, Reduce("+", nums, 0))
+  
+  rdd <- parallelize(sc, list())
+  actual <- fold(rdd, 0, "+")
+  expect_equal(actual, 0)
+})
+
+test_that("aggregate() on RDDs", {
```
Comment (concretevitamin) [inline pkg/inst/tests/test_rdd.R]: nit: "aggregateRDD()"

**Comment by concretevitamin** [comment]
Also LGTM pending a minor style comment

File: pkg/inst/tests/test_rdd.R (hunk 2 above)
Comment (sun-rui) [inline pkg/inst/tests/test_rdd.R]: fixed

File: pkg/R/RDD.R (hunk 1 above)
Comment (sun-rui) [inline pkg/R/RDD.R]: fixed.

**Comment by sun-rui** [comment]
fix comments.

**Comment by concretevitamin** [comment]
Thanks! Merging this in.

**Pull request merged**